harness/determined/cli/token.py

```diff
@@ -0,0 +1,193 @@
+    filter_data = json.dumps({"Token_Ids": args.token_id})
+    try:
+        resp = bindings.get_GetAccessTokens(session=sess, filter=filter_data)
+        if args.json or args.yaml:
+            json_data = [t.to_json() for t in resp.tokenInfo]
+            if args.json:
+                render.print_json(json_data)
+            else:
+                print(util.yaml_safe_dump(json_data, default_flow_style=False))
+        else:
+            render_token_info(resp.tokenInfo)
+    except api.errors.APIException as e:
+        raise errors.CliError(f"Caught APIException: {str(e)}")
+    except Exception as e:
+        raise errors.CliError(f"Error fetching tokens: {e}")
+
+
+def list_tokens(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    filter_data = {
+        **({"Username": args.username} if args.username else {}),
+        **({"only_Active": args.only_active} if args.only_active else {}),
+    }
+    try:
+        filter_json = json.dumps(filter_data)
+        resp = bindings.get_GetAccessTokens(sess, filter=filter_json)
+        if args.json or args.yaml:
+            json_data = [t.to_json() for t in resp.tokenInfo]
+            if args.json:
+                render.print_json(json_data)
+            else:
+                print(util.yaml_safe_dump(json_data, default_flow_style=False))
+        else:
+            render_token_info(resp.tokenInfo)
+    except Exception as e:
+        print(f"Error fetching tokens: {e}")
+
+
+def revoke_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    try:
+        request = bindings.v1PatchAccessTokenRequest(
+            tokenId=args.token_id, description=None, setRevoked=True
+        )
+        resp = bindings.patch_PatchAccessToken(sess, body=request, tokenId=args.token_id)
+        print(json.dumps(resp.to_json(), indent=2))
+    except api.errors.NotFoundException:
+        raise errors.CliError("Token not found")
+    print(f"Successfully updated token with ID: {args.token_id}.")
+
+
+def create_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    username = args.username or sess.username
+    user = bindings.get_GetUserByUsername(session=sess, username=username).user
+
+    if user is None or user.id is None:
+        raise errors.CliError(f"User '{username}' not found or does not have an ID")
+
+    request = None
+    request = bindings.v1PostAccessTokenRequest(
+        userId=user.id, lifespan=args.expiration_duration, description=args.description
+    )
+    resp = bindings.post_PostAccessToken(sess, body=request).to_json()
+
+    output_string = None
+    if args.yaml:
+        output_string = util.yaml_safe_dump(resp, default_flow_style=False)
+    elif args.json:
+        output_string = json.dumps(resp, indent=2)
+    else:
+        output_string = f'{resp["token"]}\n{resp["tokenId"]}'
+
+    if args.output_file:
+        with open(args.output_file, "w") as file:
+            file.write(output_string)
```

Comment:
do we want the token ID written to the file?

Reply:
My 2cents: yes, it would make customers to keep track of tokens easier. but I am open to go other way.

@corban-beaird : thoughts?

Reply:
I would lean against it, if folks want to use the file contents directly in the `det token login` command, they'd have to clean up the file themselves or we'd have to add additional parsing on our end to handle that extra value